pkg/scheduler/service/service.go

```diff
@@ -473,14 +473,25 @@ func (s *SchedulerServer) awaitFailover(schedulerProcess schedulerProcessInterfa
 
 func validateLeadershipTransition(desired, current string) {
 	log.Infof("validating leadership transition")
-	d := uid.Parse(desired).Group()
-	c := uid.Parse(current).Group()
-	if d == 0 {
+	desiredUID, err := uid.Parse(desired)
+	if err != nil {
+		log.Fatalf("Failed to parse desired scheduler UID %q: %v", desired, err)
```

Comment:
Would you mind adding a func-level comment (one that I should have added but did not)? something along the lines of "this func may terminate the program if leadership transition validation fails. this is purposeful and attempts to thwart split-brain scenarios. it *should* be safe to terminate this way because there should be some sort of external service monitoring process in place that will restart the scheduler after it terminates".